webapp/src/filework/mod.rs

```diff
@@ -0,0 +1,35 @@
+#![allow(unused)]
+use std::fs::{File, remove_file};
+use std::io::{Write, Seek, SeekFrom};
+use std::env;
+use std::path::{PathBuf, Path};
+
+pub fn write_source_to_file(source_code: &str, lang: &str) -> PathBuf
+{
+    // If error ever happens on work with temp files it's not on the user, 
+    // so he should get "internal server error" here
+    let mut input_file_name = env::var("COMPILATION_TEMP_DIR")
+        .expect("COMPILATION_TEMP_DIR doesn't exist");
+    input_file_name.push_str("/");
+    input_file_name.push_str(
+        &chrono::Utc::now()
+        .format("compilation_input-%Y-%m-%d-%H-%M-%S-%f").to_string()
+    );
+    input_file_name.push_str(".cpp");     // TODO: add lang specific extensions
+
+    let mut code_file = File::create(&input_file_name)
+        .expect("Panic on temp file creation");  
+    code_file.write_all(source_code.as_bytes())
+        .expect("Panic on copying of source code into the temp file");
+    code_file.seek(SeekFrom::Start(0)).expect("Panic on temp file seek");
+    
+    drop(code_file);
```

Comment:
Code file should be automatically dropped when going out of its scope, is there any need in doing this explicitly?

Reply:
I did this because I reckoned the compiler would need the file closed, but I didn't ask if it actually is so, my bad. If the compiler is okay with the file being open elsewhere then dropping it explicitly is redundant.

Reply:
Oh, I see now. Body of this function was once a part of http_handlers::submit::post_submit() and this drop migrated from there. It is completely redundant here, it will be removed.